Locker/Internal/LockerHelpers.swift

```diff
@@ -0,0 +1,226 @@
+        let userDefaultsDidAskToUseBiometricsID = "\(LockerHelpers.bundleIdentifier)_UserDefaultsDidAskToUseTouchID"
+        return "\(userDefaultsDidAskToUseBiometricsID)_\(uniqueIdentifier)"
+    }
+
+    static func keyBiometricsIDActivatedForUniqueIdentifier(_ uniqueIdentifier: String) -> String {
+        let userDefaultsKeyBiometricsIDActivated = "\(LockerHelpers.bundleIdentifier)_UserDefaultsKeyTouchIDActivated"
+        return "\(userDefaultsKeyBiometricsIDActivated)_\(uniqueIdentifier)"
+    }
+
+    static func keyShouldAddSecretToKeychainOnNextLoginForUniqueIdentifier(_ uniqueIdentifier: String) -> String {
+        let userDefaultsShouldAddSecretToKeychainOnNextLogin = "\(LockerHelpers.bundleIdentifier)_UserDefaultsShouldAddPasscodeToKeychainOnNextLogin"
+        return "\(userDefaultsShouldAddSecretToKeychainOnNextLogin)_\(uniqueIdentifier)"
+    }
+
+    // MARK: - Biometric helpers
+
+    static func storeCurrentLAPolicyDomainState() {
+        let newDomainState = LockerHelpers.currentLAPolicyDomainState
+        LockerHelpers.setLAPolicyDomainState(with: newDomainState)
+    }
+
+    static var keyKeychainServiceName: String {
+        return "\(LockerHelpers.bundleIdentifier)_KeychainService"
+    }
+}
+
+// MARK: - Private extension
+
+private extension LockerHelpers {
+
+    static func checkIfBiometricsSettingsChanged() -> Bool {
+        let oldDomainState = LockerHelpers.savedLAPolicyDomainState
+        let newDomainState = LockerHelpers.currentLAPolicyDomainState
+
+        // Check for domain state changes
+        // For deactivated biometrics, LAContext in validation will return nil
+        // storing that nil and comparing it to nil will result as `isEqual` NO
+        // even data is not actually changed.
+        let biometricsDeactivated = oldDomainState == nil || newDomainState == nil
+        let biometricSettingsDidChange = oldDomainState?.elementsEqual(newDomainState!) ?? false
+        if (biometricsDeactivated && biometricSettingsDidChange) {
+            LockerHelpers.setLAPolicyDomainState(with: newDomainState)
+            return true
+        }
+
+        return false
+    }
+
+    static func checkIfCanAuthenticateWithBiometrics() -> Bool {
+        let context = LAContext()
+        var error: NSError?
+
+        if !context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error) {
+            // When user removes all fingers for TouchID, error code will be `notEnrolled`.
+            // In that case, we want to return that device supports TouchID.
+            // In case lib is used on simulator, error code will always be `notEnrolled` and only then
+            // we want to return that biometrics is not supported as we don't know what simulator is used.
+            if let error = error, error.code == kLAErrorBiometryNotAvailable || (error.code == kLAErrorBiometryNotEnrolled && isSimulator) {
```

Comment:
Check if you can use `LAError.biometryNotAvailable.rawValue` and `LAError.biometryNotEnrolled.rawValue` here instead of this old `k...` objc constants :)

Reply:
those constants are only available from iOS 11 so we would just have even more lines of code since we would have to check the availability

Reply:
Maybe implement it as private computed property? 
That way, one day when we increase min deploy target, we can simply search for `#if available` and update obsolete code from the project :)